pkgs/development/misc/resholve/resholve-package.nix

```diff
@@ -87,10 +88,19 @@ let
     # enable below for verbose debug info if needed
     # supports default python.logging levels
     # LOGLEVEL="INFO";
+    /*
+      subshell/PS4/set -x and : command to output resholve envs
+      and invocation. Extra context makes it clearer what the
+      Nix API is doing, makes nix-shell debugging easier, etc.
+    */
     preFixup = ''
-      pushd "$out"
-      ${builtins.concatStringsSep "\n" (makeCommands solutions)}
-      popd
+      (
+        cd "$out"
+        PS4=$'\x1f'"\033[33m[resholve context]\033[0m "
+        set -x
+        : changing directory to $PWD
```

Comment:
I have a question regarding this line.
I know that the `:` operator is a no-op. I've seen it used to redefine some variables.
I don't understand what is happening on that line.
Are you adding the `:` so that the rest of the line will not be interpreted ?
The rest of the line won't be echoed though right ?

Reply:
because I'm using `set -x`, it'll show up in the trace output. Along with the `PS4` set here it ends up looking like:

```console
[resholve context] : changing directory to /nix/store/f1kc91yzrz5iwhagg9nigp0fwqy4a4j7-shunit2-2019-08-10
```

Reply:
Clever! Thank you!